src/serializer/modules.js

```diff
@@ -207,6 +213,7 @@ export class ModuleTracer extends Tracer {
 
             if (effects === undefined) {
               console.log(`delaying require(${moduleIdValue})`);
+              if (this.logStatistics) this.statistics.delayedModules++;
```

Comment:
In this case, when we are delaying, we are throwing away any partial effects computed by `evaluateForEffects` above, and in that case, we should not keep around statistics about abandoned nested delayed modules.

Meaning: Please save `this.statistics.delayedModules` before the call to `evaluateForEffects`, and if we get into this `effects === undefined` case, restore the value before  incrementing it.